### PR TITLE
libbpf-tools: add runqlat

### PR DESCRIPTION
**Pull request by ethercflow**
Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### CODE REVIEW

**Comment by ethercflow** [comment]
Hi @anakryiko, another tool PTAL when you have time, thank you. :)

File: libbpf-tools/runqlat.c [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+		break;
+	case 'L':
+		env.per_thread = true;
+		break;
+	case 'P':
+		env.per_process = true;
+		break;
+	case OPT_PIDNSS:
+		env.per_pidns = true;
+		break;
```
Comment (anakryiko) [inline libbpf-tools/runqlat.c:90]: these seem to be mutually exclusive, can you please enforce that?

Comment (ethercflow) [reply libbpf-tools/runqlat.c]: Fixed, thanks!

**Comment by ethercflow** [comment]
Besides mutually exclusive, I modified another three places：
1.  change to `print_log2_hists(struct bpf_map *hists)` instead of passing `fd`;
2.  change to `const char *units = env.milliseconds ? "msecs" : "usecs";` instead of `char *`;
3.  change `void print_log2_hist(unsigned int *vals, int vals_size, const char *val_type);` 's thrid  param type add `const`;

Thank you.

File: libbpf-tools/runqlat.c [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		if (env.per_pidns || env.per_thread) {
+			fprintf(stderr, "pidnss, pids, tids cannot be used "
+				"together\n");
+			argp_usage(state);
+		}
+		env.per_process = true;
+		break;
+	case OPT_PIDNSS:
+		if (env.per_process || env.per_thread) {
+			fprintf(stderr, "pidnss, pids, tids cannot be used "
```
Comment (anakryiko) [inline libbpf-tools/runqlat.c:100]: I'd just do this in one place, after args are parsed, no need to triplicate all this

**Review by anakryiko — APPROVED** [review]
looks good, consider checking args in main(), after arguments are parsed

**Comment by ethercflow** [comment]
Hi @yonghong-song , PTAL when you have time, thanks! :)

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request merged**